Modules/_xxtestfuzz/fuzzer.c

```diff
@@ -501,6 +501,60 @@ static int fuzz_elementtree_parsewhole(const char* data, size_t size) {
     return 0;
 }
 
+#define MAX_PYCOMPILE_TEST_SIZE 16384
+static char pycompile_scratch[MAX_PYCOMPILE_TEST_SIZE];
+
+static const int start_vals[] = {Py_eval_input, Py_single_input, Py_file_input};
+const size_t NUM_START_VALS = sizeof(start_vals) / sizeof(start_vals[0]);
+
+static const int optimize_vals[] = {-1, 0, 1, 2};
+const size_t NUM_OPTIMIZE_VALS = sizeof(optimize_vals) / sizeof(optimize_vals[0]);
+
+/* Fuzz `PyCompileStringExFlags` using a variety of input parameters.
+ * That function is essentially behind the `compile` builtin */
+static int fuzz_pycompile(const char* data, size_t size) {
+    if (size > sizeof(pycompile_scratch)) {
+        return 0;
+    }
+
+    // Need 2 bytes for parameter selection plus 1 for null terminator
+    if (size < 2 + 1)
```

Comment:
We don't need a nul terminator to come from the fuzzer, I think this can just be < 2?

Reply:
On line 534, you'll see that the `pycompile_scratch` (a modified copy of the input buffer) is `nul`-terminated. The `Py_CompileStringExFlags` function expects a `nul`-terminated C string as input. I don't think the fuzzer is guaranteed to give us a such an input?

Reply:
It's not, but you're already copying data into `pycompile_scratch` -- that means you can just allocate an extra byte in it for the `NUL`; there's no need for that to be dependent on the input size.

Reply:
Yes, that's a good point. I'll change this.

Reply:
This should be fixed now.